tests/phpunit/Civi/Funding/FundingCase/Handler/FundingCaseRecipientContactSetHandlerTest.php

```diff
@@ -0,0 +1,142 @@
+      $this->possibleRecipientsLoaderMock
+    );
+  }
+
+  public function testHandle(): void {
+    $command = $this->createCommand();
+    $this->actionsDeterminerMock->method('isActionAllowed')
+      ->with(
+        'set-recipient-contact',
+        $command->getFundingCase()->getStatus(),
+        $command->getApplicationProcessStatusList(),
+        $command->getFundingCase()->getPermissions()
+      )
+      ->willReturn(TRUE);
+
+    $this->possibleRecipientsLoaderMock->method('getPossibleRecipients')
+      ->with($command->getFundingCase()->getCreationContactId(), $command->getFundingProgram())
+      ->willReturn([1234 => 'New Recipient']);
+
+    $this->fundingCaseManagerMock->expects(static::once())->method('update')
+      ->with($command->getFundingCase());
+
+    $this->handler->handle($command);
+    static::assertSame(1234, $command->getFundingCase()->getRecipientContactId());
+  }
+
+  public function testHandleNotInPossibleContacts(): void {
+    $command = $this->createCommand();
+    $this->actionsDeterminerMock->method('isActionAllowed')
+      ->with(
+        'set-recipient-contact',
+        $command->getFundingCase()->getStatus(),
+        $command->getApplicationProcessStatusList(),
+        $command->getFundingCase()->getPermissions()
+      )
+      ->willReturn(TRUE);
+
+    $this->possibleRecipientsLoaderMock->method('getPossibleRecipients')
+      ->with($command->getFundingCase()->getCreationContactId(), $command->getFundingProgram())
+      ->willReturn([$command->getRecipientContactId() + 1 => 'Some Recipient']);
+
+    $this->expectException(\InvalidArgumentException::class);
+    $this->expectExceptionMessage('Invalid recipient contact ID');
+    $this->handler->handle($command);
+  }
+
+  public function testHandleUnauthorized(): void {
+    $command = $this->createCommand();
+    $this->actionsDeterminerMock->method('isActionAllowed')
+      ->with(
+        'set-recipient-contact',
+        $command->getFundingCase()->getStatus(),
+        $command->getApplicationProcessStatusList(),
+        $command->getFundingCase()->getPermissions()
+      )
+      ->willReturn(FALSE);
+
+    $this->expectException(UnauthorizedException::class);
+    $this->expectExceptionMessage('Changing the recipient contact of this funding case is not allowed.');
+    $this->handler->handle($command);
+  }
+
+  private function createCommand(): FundingCaseRecipientContactSetCommand {
+    $fundingCase = FundingCaseFactory::createFundingCase(['creation_contact_id' => 1, 'recipient_contact_id' => 2]);
+    $recipientContactId = 1234;
+    $fundingCaseType = FundingCaseTypeFactory::createFundingCaseType();
+    $fundingProgram = FundingProgramFactory::createFundingProgram();
+
+    return new FundingCaseRecipientContactSetCommand(
+      $fundingCase,
+      $recipientContactId,
+      [22 => new FullApplicationProcessStatus('eligible', TRUE, TRUE)],
+      $fundingCaseType,
+      $fundingProgram
+    );
```

Comment:
would it make sense to move this method into a separate factory for creating commands with particular test data?